src/main/java/org/folio/service/uc/UCAuthServiceImpl.java

```diff
@@ -94,6 +100,17 @@ private CompletableFuture<String> loadToken(String tenantId) {
       .thenApply(UCAuthToken::getAccessToken);
   }
 
+  private CompletableFuture<DbUCCredentials> getUCCredentials(String tenantId) {
+    return findUCCredentials(tenantId)
+      .thenApply(dbUCCredentials -> {
+        if (dbUCCredentials.isEmpty()) {
+          throw new UcAuthenticationException(INVALID_CREDENTIALS_MESSAGE);
+        } else {
+          return dbUCCredentials.get();
+        }
+      });
+  }
+
```

Comment:
You can use this new method in loadToken() method to avoid code duplicates
```
  private CompletableFuture<String> loadToken(String tenantId) {
    return getUCCredentials(tenantId)
      .thenCompose(this::requestToken)
      .thenApply(UCAuthToken::getAccessToken);
  }
```